capstone/src/main/java/com/google/sps/servlets/ChallengeServlet.java

```diff
@@ -0,0 +1,64 @@
+package com.google.sps.servlets;
+
+import java.io.IOException;
+import java.sql.Timestamp;
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.gson.Gson;
+import com.google.sps.Objects.Challenge;
+
+@WebServlet("challenge")
+public class ChallengeServlet extends HttpServlet {
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    Query query = new Query("Challenge").addSort("dueDate", SortDirection.DESCENDING);
+    PreparedQuery results = datastore.prepare(query);
+    // Get most recent challenge in database
+    Entity entity = results.asIterable().iterator().next();
+    String challengeName = (String) entity.getProperty("name");
+    long dueDate = (long) entity.getProperty("dueDate");
+    ArrayList<String> usersCompleted = (ArrayList<String>) entity.getProperty("usersCompleted");
+    // setting badge as null for now
+    Challenge challenge = new Challenge(challengeName, dueDate, null, usersCompleted);
```

Comment:
To avoid accessing properties and casting items in place, could I request in `Challenge.java`:

```java
public Challenge fromEntity(Entity) {
  ...
}
```